backend/database.js

```diff
@@ -52,38 +52,52 @@ export default ((sbp('sbp/selectors/register', {
     }
     // Number of entries pushed.
     let counter = 0
-    let currentHash = await sbp('chelonia.db/get', `_private_hidx=${contractID}#${height}`)
-    let prefix = '['
+    let { currentHash, ...serverMeta } = (
+      await sbp('chelonia/db/getEntryMeta', contractID, height)
+    ) || {}
+    let prefix = ''
     let ended = false
     // NOTE: if this ever stops working you can also try Readable.from():
     // https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options
     const stream = new Readable({
+      construct (cb) {
+        this.push('[')
+        cb()
+      },
       read (): void {
         if (ended) {
           this.destroy()
           return
         }
-        if (currentHash && counter < limit) {
+        if (!!currentHash && counter < limit) {
```

Comment:
Is this `!!` really needed? (Why?)

Reply:
No, it's not really needed (as in it behaves exactly the same with or without), more of a style choice (trying to be explicit where possible, this makes it explicit it's coerced into a boolean).

Let me know if I should remove it.

Reply:
I mean, if you're asking for my preference, I prefer it the way it was before, because if it works just the same, we save a couple opcodes (I'm neurotic like that) and it seems more readable to me personally

Reply:
> we save a couple opcodes (I'm neurotic like that) and it seems more readable to me personally

I'll remove it then if it's your preference, but, personally, I disagree that it saves opcodes or that it's more readable (if you know what `!!` does).

1. When implementing a JS interpreter, you'd need to check the type and see if it's truthy or falsy here. Whether you end up saving opcodes or not is highly speculative.
2. Readability is highly subjective, but I'd argue that `!!x` requires less cognitive context than `x` for a non-boolean type.

Reply:
> When implementing a JS interpreter, you'd need to check the type and see if it's truthy or falsy here. Whether you end up saving opcodes or not is highly speculative.

Interesting. Didn't know that!